### DISCUSSION TITLE
How to use existing SSL certificates with PocketBase?

### DISCUSSION THREAD

**Discussion by jaejungkim** [pocketbase/pocketbase#4094] [Q&A] [answered]
Hello,

I am currently using PocketBase and I have a question regarding SSL certificates. I have already purchased SSL certificates (fullchain.pem and privkey.pem) with a validity of 1 year. I would like to use these existing certificates with PocketBase instead of using the autotls feature which automatically obtains certificates from Let’s Encrypt.

Could you please guide me on how to configure PocketBase to use these existing certificates? Any help would be greatly appreciated.

Thank you in advance for your time and assistance.

**Comment by ganigeorgiev** [comment, marked as answer]
It is not possible at the moment to load custom certificate with the prebuilt binary (at least not easily). An option could be to extend PocketBase with Go and modify the `e.Server.TlsConfig` in the [`OnBeforeServe()`](https://pocketbase.io/docs/go-event-hooks/#onbeforeserve) hook.

My recommendation for now is if you want to use custom certificate and/or other more advanced network configurations is to [deploy PocketBase behind a reverse proxy](https://pocketbase.io/docs/going-to-production/#using-reverse-proxy) (nginx, apache, caddy, etc.)

**Reply by jaejungkim** [reply]
Thank you for your prompt and detailed response. Your recommendation to use a reverse proxy for advanced network configurations makes sense. I appreciate your help and will consider extending PocketBase with Go as suggested. Thanks again!
